Firestore/Example/Tests/Core/FSTQueryTests.mm

```diff
@@ -55,30 +60,31 @@ @implementation FSTQueryTests
 - (void)testConstructor {
   FSTResourcePath *path =
       [FSTResourcePath pathWithSegments:@[ @"rooms", @"Firestore", @"messages", @"0001" ]];
-  FSTQuery *query = [FSTQuery queryWithPath:path];
+  FSTQuery *query = [FSTQuery queryWithPath:[path toCPPResourcePath]];
   XCTAssertNotNil(query);
 
   XCTAssertEqual(query.sortOrders.count, 1);
-  XCTAssertEqualObjects(query.sortOrders[0].field.canonicalString, kDocumentKeyPath);
+  XCTAssertEqual(query.sortOrders[0].field.CanonicalString(), FieldPath::kDocumentKeyPath);
   XCTAssertEqual(query.sortOrders[0].ascending, YES);
 
   XCTAssertEqual(query.explicitSortOrders.count, 0);
 }
 
 - (void)testOrderBy {
   FSTQuery *query = FSTTestQuery(@"rooms/Firestore/messages");
-  query =
-      [query queryByAddingSortOrder:[FSTSortOrder sortOrderWithFieldPath:FSTTestFieldPath(@"length")
-                                                               ascending:NO]];
+  query = [query
+      queryByAddingSortOrder:[FSTSortOrder
+                                 sortOrderWithFieldPath:[FSTTestFieldPath(@"length") toCPPFieldPath]
```

Comment:
In the end state of this C++ conversion all of these test helpers (`FSTTestFieldPath` here) should be ported too. All of these are too long in Objective-C, so it's worth looking at how this works in the Android or JS ports.

On Android a lot of these are grouped in `TestUtil`. This one is `TestUtil.field(String)`. On JS this is just a [free function named Field](https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/test/util/helpers.ts#L162).

On Android we have static imports, and in JS we have ES6 imports both of which allow us to import `field` to look like a free function, keeping the testing code compact. In C++ there's no easy way to alias a function so we won't quite be able to make the equivalent.

I've already started created some unit testing utilities in a [C++ testutil](https://github.com/firebase/firebase-ios-sdk/tree/master/Firestore/core/test/firebase/firestore/testutil).

I suggest:
  * Create `Firestore/core/test/firebase/firestore/testutil/testutil.h`
  * Namespace `firebase::firestore::testutil`
  * Free functions just within there, probably with names equivalent to the js-sdk, not objc

Then create a namespace alias locally in the tests that use helpers. We could even make that helper shorter than `testutil` given all the usage in here. Maybe even `namespace t = firebase::firestore::testutil`? Dunno if that last part is a good idea.